### PR TITLE
Move nsec backup to drawer and organize drawer

### PR DESCRIPTION
**Pull request by maxmoney21m**
Regarding the first step mentioned in #59, this PR moves the nsec backup from the profile page to a separate screen using a Dialog in the drawer. I also organized the Drawer slightly and added some icons. See screenshots for reference.

There is no biometric check yet, but if the PR is OK so far I can add that.

image

image

image

### CODE REVIEW

File: app/src/main/java/com/vitorpamplona/amethyst/ui/navigation/Routes.kt [hunk 1]
```diff
@@ -53,7 +53,7 @@ sealed class Route(
         buildScreen = { acc, accSt, nav -> { _ -> ChatroomListScreen(acc, nav) }}
     )
 
-    object Filters : Route("Filters", R.drawable.ic_dm,
+    object Filters : Route("Filters", R.drawable.ic_security,
```
Comment (maxmoney21m) [inline app/src/main/java/com/vitorpamplona/amethyst/ui/navigation/Routes.kt:56]: Added a shield icon

File: app/src/main/java/com/vitorpamplona/amethyst/ui/navigation/DrawerContent.kt [hunk 2]
```diff
@@ -206,66 +214,84 @@ fun ListContent(
...
     val currentRoute = currentRoute(navController)
+    IconRow(title, icon, tint, onClick = {
+        if (currentRoute != route) {
+            navController.navigate(route)
+        }
+        coroutineScope.launch {
+            scaffoldState.drawerState.close()
+        }
+    })
+}
```
Comment (maxmoney21m) [inline app/src/main/java/com/vitorpamplona/amethyst/ui/navigation/DrawerContent.kt:288]: Extracted out `IconRow` from `NavigationRow` for reuse.

File: app/src/main/java/com/vitorpamplona/amethyst/ui/navigation/DrawerContent.kt [hunk 3]
```diff
@@ -206,66 +214,84 @@ fun ListContent(
...
     modifier: Modifier,
-    accountViewModel: AccountStateViewModel
+    accountViewModel: AccountStateViewModel,
+    account: Account,
 ) {
     val coroutineScope = rememberCoroutineScope()
+    var backupDialogOpen by remember { mutableStateOf(false) }
 
-    Column(modifier = modifier) {
-        LazyColumn() {
```
Comment (maxmoney21m) [inline app/src/main/java/com/vitorpamplona/amethyst/ui/navigation/DrawerContent.kt:214]: I'm not sure why this was previously `Column() { LazyColumn() { Column() }}}` It seems to work fine with just a single `Column`. Can revert if I'm missing something...

File: app/src/main/java/com/vitorpamplona/amethyst/ui/screen/loggedIn/AccountBackupDialog.kt [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+                    NSecCopyButton(account)
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun NSecCopyButton(
```
Comment (maxmoney21m) [inline app/src/main/java/com/vitorpamplona/amethyst/ui/screen/loggedIn/AccountBackupDialog.kt:91]: Function copied from original in `ProfileScreen.kt`

File: app/src/main/java/com/vitorpamplona/amethyst/ui/screen/loggedIn/ProfileScreen.kt [hunk 5]
```diff
@@ -637,40 +631,7 @@ fun TabRelays(user: User, accountViewModel: AccountViewModel, navController: Nav
     }
 }
 
-@Composable
-private fun NSecCopyButton(
```
Comment (maxmoney21m) [inline app/src/main/java/com/vitorpamplona/amethyst/ui/screen/loggedIn/ProfileScreen.kt:641]: Removed and placed in `AccountBackupDialog.kt`

File: app/src/main/java/com/vitorpamplona/amethyst/ui/navigation/DrawerContent.kt [hunk 6]
```diff
@@ -206,66 +214,84 @@ fun ListContent(
...
+        )
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        IconRow(
+            "Logout",
+            R.drawable.ic_logout,
+            MaterialTheme.colors.onBackground,
+            onClick = { accountViewModel.logOff() }
+        )
```
Comment (maxmoney21m) [inline app/src/main/java/com/vitorpamplona/amethyst/ui/navigation/DrawerContent.kt:261]: The `Spacer` with `weight(1f)` causes the items below to go to the bottom of the `Column`. Moved the `Logout` button here.

File: app/src/main/java/com/vitorpamplona/amethyst/ui/navigation/DrawerContent.kt [hunk 7]
```diff
@@ -206,66 +214,84 @@ fun ListContent(
...
+            title = "Backup Keys",
+            icon = R.drawable.ic_key,
+            tint = MaterialTheme.colors.onBackground,
+            onClick = { backupDialogOpen = true }
+        )
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        IconRow(
+            "Logout",
```
Comment (maxmoney21m) [inline app/src/main/java/com/vitorpamplona/amethyst/ui/navigation/DrawerContent.kt:257]: To replace with string resource.

File: app/src/main/java/com/vitorpamplona/amethyst/ui/screen/loggedIn/AccountBackupDialog.kt [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+        modifier = Modifier.padding(horizontal = 3.dp),
+        onClick = {
+            account.loggedIn.privKey?.let {
+                clipboardManager.setText(AnnotatedString(it.toNsec()))
+                scope.launch {
+                    Toast.makeText(
+                        context,
+                        context.getString(R.string.secret_key_copied_to_clipboard),
+                        Toast.LENGTH_SHORT
+                    ).show()
```
Comment (maxmoney21m) [inline app/src/main/java/com/vitorpamplona/amethyst/ui/screen/loggedIn/AccountBackupDialog.kt:108]: Popup a `Toast` to inform the user that it was successfully copied.

**Comment by vitorpamplona** [comment]
Love it!

**Comment by vitorpamplona** [comment]
What's your npub, @maxmoney21m ?

**Comment by braydonf** [comment]
Nice work! It would be great to add mnemonic phrase here as well at some point too.

**Comment by maxmoney21m** [comment]
@vitorpamplona npub1max2lm5977tkj4zc28djq25g2muzmjgh2jqf83mq7vy539hfs7eqgec4et

**Pull request merged**